### ISSUE TITLE
Completion for  command prefixed environment variables

### ISSUE THREAD

**Issue by gavenkoa** [scop/bash-completion#452]
I find it cumbersome to type `JAVA_HOME`:

```
JAVA_HOME=/... mvn
```

and the most annoying thing is undefined env vars. Like I have to use `ANSIBLE_CONFIG` frequently and it is not defined and I forgot exact name.

Another surprise: there is no completion for `env` (( It might be the easiest way to implement this feature initially.

**Comment by scop** [comment]
I can see the value in that, but I'm not convinced we want to go into maintaining a list of "known good" environment variable names to suggest when one might be applicable but is not defined.

Not sure what you mean by no completion for `env`, there sure is one. If you mean it does not complete env variable names, that's right. Making it complete defined ones would likely be doable, but completing names of undefined ones falls into the same category as above.

**Comment by gavenkoa** [comment]
> maintaining a list of "known good" environment variable names

It should be another environment variable holding space separated list of names. Or point to file or use predefined file path with single name per line. This feature will be useless if you cannot alter the list for your personal taste.

There is no problem with `env` or alternatively named wrapper. I can write completion independently.

It is interesting to have completion for environ. vars when you type regular commands: `ANSIBLE_CONFIG=my.yml ansible -i ...`

**Comment by gavenkoa** [comment]
> Not sure what you mean by no completion for env, there sure is one. 

Yes, it is almost useless:

```
# complete -p | grep env
complete -F _longopt env
```

That's why I considered it missing: there is no completion for command:

```
env [OPTION]... [-] [NAME=VALUE]... [COMMAND [ARG]...]
```

What is the value for completion for `--help`?

**Comment by scop** [comment]
Yes, the env completion could be improved quite a bit. Ditto many other completions. If your intent is to inspire others to write code for the use cases you care about, I think the tone of the comments so far is not likely to contribute towards that. How about filing a PR instead? :)
